packages/stryker/src/reporters/ScoreReporter.ts

```diff
@@ -0,0 +1,70 @@
+import {Reporter, ScoreResult} from 'stryker-api/report';
+import StrykerBadgeClient from './ScoreReporter/StrykerBadgeClient';
+import {getEnvironmentVariable} from '../utils/objectUtils';
+import { getLogger } from 'log4js';
+import { StrykerOptions } from 'stryker-api/core';
+
+export default class ScoreReporter implements Reporter {
+  private readonly log = getLogger(ScoreReporter.name);
+
+  constructor(
+    setting: StrykerOptions,
+    private strykerBadgeClient: StrykerBadgeClient = new StrykerBadgeClient()
+  ) { }
+
+  private isPullRequest: boolean;
+  private repository: string | undefined;
+  private branch: string | undefined;
+  private apiKey: string | undefined;
+  private mutationScore: number;
+
+  private readEnvironmentVariables() {
```

Comment:
This method has some code duplication. Could we move the reading of env variables and logging of a warning to a private function?

Reply:
Done